storage/container.go

```diff
@@ -0,0 +1,178 @@
+package storage
+
+import (
+	"net/http"
+	"strconv"
+	"strings"
+)
+
+const CONTAINER_VERSION = "v1"
+
+// Container describes an existing Container.
+type Container struct {
+	// The name of the Container
+	Name string
+	// A container access control list (ACL) that grants read access.
+	ReadACLs []string
+	// A container access control list (ACL) that grants write access
+	WriteACLs []string
+	// The secret key value for temporary URLs.
+	URLKey string
+	// The second secret key value for temporary URLs.
+	URLKey2 string
+	// List of origins to be allowed to make cross-origin Requests.
+	AllowedOrigins []string
+	// Maximum age in seconds for the origin to hold the preflight results.
+	MaxAge int
+}
+
+// CreateContainerInput defines an Container to be created.
+type CreateContainerInput struct {
+	// The unique name for the container. The container name must be from 1 to 256 characters long and can
+	// start with any character and contain any pattern. Character set must be UTF-8. The container name
+	// cannot contain a slash (/) character because this character delimits the container and object name.
+	// For example, /account/container/object.
+	// Required
+	Name string `json:"name"`
+	// Sets a container access control list (ACL) that grants read access.
+	ReadACLs []string
+	// Sets a container access control list (ACL) that grants read access.
+	WriteACLs []string
+	// Sets a secret key value for temporary URLs.
+	URLKey string
+	// Sets a second secret key value for temporary URLs.
+	URLKey2 string
+	// Sets the list of origins allowed to make cross-origin requests.
+	AllowedOrigins []string
+	// Sets the maximum age in seconds for the origin to hold the preflight results.
+	MaxAge int
+}
+
+// CreateContainer creates a new Container with the given name, key and enabled flag.
+func (c *StorageClient) CreateContainer(createInput *CreateContainerInput) (*Container, error) {
+	headers := make(map[string]string)
+
+	createInput.Name = c.getQualifiedName(CONTAINER_VERSION, createInput.Name)
+
+	// There are default values for these that we don't want to zero out if Read and Write ACLs are not set.
+	if len(createInput.ReadACLs) > 0 {
+		headers["X-Container-Read"] = strings.Join(createInput.ReadACLs, ",")
+	}
+	if len(createInput.WriteACLs) > 0 {
+		headers["X-Container-Write"] = strings.Join(createInput.WriteACLs, ",")
+	}
+
+	headers["X-Container-Meta-Temp-URL-Key"] = createInput.URLKey
+	headers["X-Container-Meta-Temp-URL-Key-2"] = createInput.URLKey2
+	headers["X-Container-Meta-Access-Control-Expose-Headers"] = strings.Join(createInput.AllowedOrigins, " ")
+	headers["X-Container-Meta-Access-Control-Max-Age"] = strconv.Itoa(createInput.MaxAge)
+
+	if err := c.createResource(createInput.Name, headers); err != nil {
```

Comment:
Does the raw `PUT` request for creation return the created container? If so, we can return that from the createResource method, and return that here. The provider should also call a subsequent Read, so by the Create method making a subsequent `GET` request, we'll be duplicating requests on every create.